crates/weaver_live_check/README.md

```diff
@@ -0,0 +1,239 @@
+# Weaver Live Check
+
+Live check is a developer tool for assessing sample telemetry and providing advice for improvement.
+
+A Semantic Convention `Registry` is loaded for comparison with samples. `Ingesters` transform various input formats and sources into intermediary representations to be assessed by `Advisors`. The `Advice` produced is transformed via jinja templates to the required output format for downstream consumption.
+
+```mermaid
+flowchart LR
+    subgraph Inputs
+        file["File"]
+        stdin["stdin"]
+        otlp["OTLP"]
+    end
+
+    subgraph Core["Processing"]
+        registry["Registry"]
+        ingesters["Ingesters"]
+
+        subgraph advisors["Advisors"]
+            builtin["Builtin"]
+
+            subgraph external["External"]
+                otel["Otel"]
+                custom["Custom"]
+            end
+        end
+    end
+
+    subgraph Outputs
+        advice["Advice"]
+        templates["Jinja Templates"]
+        output["Output Format"]
+    end
+
+    file --> ingesters
+    stdin --> ingesters
+    otlp --> ingesters
+
+    registry -- "Loaded for comparison" --> advisors
+    ingesters -- "Intermediary representations" --> advisors
+
+    builtin --> advice
+    external --> advice
+
+    advice -- "Transformed via" --> templates
+    templates --> output
+```
+
+## Ingesters
+
+Sample data can have various levels of detail; from a simple list of attribute names, to a full OTLP signal structure. This data can come from different sources: files, stdin, OTLP. Therefore you need to choose the appropriate `Ingester` for your job:
+
+- `attribute_file | AF | af` => for attribute name or name=value pair text files
+- `attribute_stdin | AS | as` => for attribute name or name=value pairs on stdin
+- `attribute_json_file | AJF | ajf` => for an array of attribute samples in a json file
+- `attribute_json_stdin | AJS | ajs` => for an array of attribute samples in json on stdin
+- `attribute_otlp | AO | ao` => for attributes extracted from OTLP signals
+
+Some `Ingesters`, like `attribute_stdin` and `attribute_otlp`, can stream the input data so you receive output at the command line as it comes in. This is really useful in live debugging sessions allowing you to breakpoint, step through your code and see live assessment as the data is received in Weaver.
+
+## Advisors
+
+Sample entities are assessed by the set of `Advisors` to produce a list of `Advice` for each one. Built-ins check for fundamental compliance with the `Registry` supplied, for example `missing_attribute` and `type_mismatch`.
+
+Beyond the fundamentals, external `Advisors` can be defined in Rego policies. The OpenTelemetry Semantic Conventions rules are included out-of-the-box by default. They provide `Advice` on name-spacing and formatting aligned with the standard. These default policies can be overridden at the command line with your own.
```

Comment:
Are we going to provide semantic convention rules out of the box by default?

(a) I'm not sure all semantic convention rules actually are useful for general projects.
(b) A *subset* of semantic convention rules (or semantic convention usage rules) would be useful to assert.

I haven't read through what you have in this PR yet, so I'll update this comment if I see something along the lines of (a).  If you don't have any out of the box rules yet, we should mark this aspirational.